### ISSUE TITLE
Test on Windows

### ISSUE THREAD

**Issue by giordano** [UCL-ARC/arc-drs-2024-julia-workshop#2]
I tested the notebook on both Linux and macOS, with both the local and Docker container methods.  I have no way to test it on Windows though, it'd be good to make sure it runs smoothly also on Windows with both methods.
```[tasklist]
### Tasks
- [x] Check instructions for running notebook locally work on Windows
- [x] Check instructions for running notebook in Docker container works on Windows
```

**Comment by matt-graham** [comment]
Everything worked without any issue following instructions for running notebook locally on Windows using both PowerShell and CMD prompts.

I installed Julia using `winget` method 
```
winget install julia -s msstore
```
I already had Git installed but [looks like](https://winget.run/pkg/Git/Git) we can also use `winget` to install that too with
```
winget install -e --id Git.Git
```
I got a message about Pluto notebook running initially in safe mode which I hadn't seen before, but I assume this a new Pluto feature rather than something specific to Windows (and was fairly obvious to deal with either way).

**Comment by giordano** [comment]
> I got a message about Pluto notebook running initially in safe mode which I hadn't seen before, but I assume this a new Pluto feature rather than something specific to Windows (and was fairly obvious to deal with either way).

Yes, the "safe mode" is a standard feature, when you open the notebook is always "inactive"

**Comment by matt-graham** [comment]
For the Docker approach, after [installing Docker (Desktop) with `winget`](https://winget.run/pkg/Docker/DockerDesktop) I could run
```
docker pull ghcr.io/ucl-arc/julia-workshop:main
```
to pull the image without any issues.

On trying to run the `docker run` command I however hit a few snags due to both PowerShell and CMD prompt seeming to have weird ways of dealing with string quoting (and at least for PowerShell not corresponding to my reading of [their documentation on this](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_quoting_rules?view=powershell-5.1#including-quote-characters-in-a-string)). Running the provided

```
docker run -p 1234:1234 -ti ghcr.io/ucl-arc/julia-workshop:main julia -e 'using Pluto; Pluto.run(; host="0.0.0.0", port=1234)'
```

in PowerShell I get a parse error from Julia

```
ERROR: ParseError:
# Error @ none:1:31
using Pluto; Pluto.run(; host=0.0.0.0, port=1234)
#                             └──┘ ── invalid numeric constant
```

which suggested that internal double quotation marks had been removed. Trying to escape them by doubling

```
docker run -p 1234:1234 -ti ghcr.io/ucl-arc/julia-workshop:main julia -e 'using Pluto; Pluto.run(; host=""0.0.0.0"", port=1234)'
```

I then get another Julia error

```
# Error @ none:1:40
using Pluto; Pluto.run(; host="0.0.0.0,
#                                      └ ── unterminated string literal
```

I finally got things to work by doubling again the final quotation marks

```
docker run -p 1234:1234 -ti ghcr.io/ucl-arc/julia-workshop:main julia -e 'using Pluto; Pluto.run(; host=""0.0.0.0"""", port=1234)'
```

though doubling both initial and final also works

```
docker run -p 1234:1234 -ti ghcr.io/ucl-arc/julia-workshop:main julia -e 'using Pluto; Pluto.run(; host=""""0.0.0.0"""", port=1234)'
```

😕 

Sadly neither of the above works in CMD prompt 🤦, however 

```
docker run -p 1234:1234 -ti ghcr.io/ucl-arc/julia-workshop:main julia -e "using Pluto; Pluto.run(; host=""0.0.0.0"""", port=1234)"
```

and

```
docker run -p 1234:1234 -ti ghcr.io/ucl-arc/julia-workshop:main julia -e "using Pluto; Pluto.run(; host=""""0.0.0.0"""", port=1234)"
```

both do. I can't find any combination however that works on both PowerShell and CMD 🙄.

Given that both PowerShell and CMD should be installed by default providing participants aren't on very old (pre Windows 7) versions, I think probably just saying to use one and giving the relevant command for that is probably the best approach.

One other snag I hit against was that the `http://0.0.0.0:1234/?secret=...` URL outputed by Pluto does not seem to work on any of the browsers I tested on Windows (Firefox, Chrome, Edge). Replacing `0.0.0.0` with `localhost` does work however.

**Comment by dpshelio** [comment]
what about using double quote everywhere and escaping the host double quote like this: `"using Pluto; Pluto.run(; host=\"0.0.0.0\", port=1234)"`?

**Comment by matt-graham** [comment]
I don't think `\` escapes in PowerShell, at least when running
```PowerShell
docker run -p 1234:1234 -ti ghcr.io/ucl-arc/julia-workshop:main julia -e "using Pluto; Pluto.run(; host=\"0.0.0.0\", port=1234)"
```
I get a Julia error
```
ERROR: ParseError:
# Error @ none:1:40
using Pluto; Pluto.run(; host=" 0.0.0.0\,
#                                      └┘ ── invalid escape sequence
```
which suggests the `\` was passed through.

Interestingly your suggestion does work in CMD prompt however.

The [PowerShell documentation on quoting](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_quoting_rules?view=powershell-5.1#including-quote-characters-in-a-string) suggests we should be able to use a backtick to escape however 

```
docker run -p 1234:1234 -ti ghcr.io/ucl-arc/julia-workshop:main julia -e "using Pluto; Pluto.run(; host=`"0.0.0.0`", port=1234)"
```

errors with

```
ERROR: ParseError:
# Error @ none:1:31
using Pluto; Pluto.run(; host=0.0.0.0, port=1234)
#                             └──┘ ── invalid numeric constant
```

Based on examples [here](https://www.rlmueller.net/PowerShellEscape.htm) I tried escaping the backtick with a backslash and that does work, that is

```
docker run -p 1234:1234 -ti ghcr.io/ucl-arc/julia-workshop:main julia -e "using Pluto; Pluto.run(; host=\`"0.0.0.0\`", port=1234)"
```

In conclusion string quoting / escaping in PowerShell is a bit of a mess 🙃.

**Comment by matt-graham** [comment]
The updates to README in #4 included adding some details on the couple of places where Windows specific commands needed to be used, so I think this can be closed, but feel free to reopen @giordano if you think there is any outstanding.
